extensions/vertx-http/runtime/src/main/java/io/quarkus/vertx/http/runtime/handlers/DevClasspathStaticHandler.java

```diff
@@ -0,0 +1,164 @@
+package io.quarkus.vertx.http.runtime.handlers;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.net.URL;
+import java.util.Set;
+
+import org.jboss.logging.Logger;
+
+import io.quarkus.vertx.http.runtime.GeneratedStaticResourcesRecorder;
+import io.vertx.core.Handler;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.http.HttpHeaders;
+import io.vertx.core.http.HttpMethod;
+import io.vertx.core.http.impl.MimeMapping;
+import io.vertx.ext.web.RoutingContext;
+import io.vertx.ext.web.handler.StaticHandler;
+
+/**
+ * {@link StaticHandler} implementation to handle static resources using the Classpath.
+ * This is meant to be used on {@link io.quarkus.runtime.LaunchMode#DEVELOPMENT} mode.
+ */
+public class DevClasspathStaticHandler implements Handler<RoutingContext> {
+
+    private static final Logger LOG = Logger.getLogger(DevClasspathStaticHandler.class);
+    private static final Set<HttpMethod> ALLOWED_HTTP_METHODS = Set.of(HttpMethod.GET, HttpMethod.HEAD, HttpMethod.OPTIONS);
+    private static final int HTTP_STATUS_OK = 200;
+    private static final String ALLOW_HEADER = "Allow";
+    private static final String ALLOW_HEADER_VALUE = "HEAD,GET,OPTIONS";
+    private final Set<String> generatedResources;
+    private final Set<String> compressedMediaTypes;
+    private final ClassLoader currentClassLoader;
+    private final boolean enableCompression;
+    private final String indexPage;
+    private final boolean enableRangeSupport;
+
+    public DevClasspathStaticHandler(Set<String> generatedResources, GeneratedStaticResourcesHandlerOptions options) {
+        this.generatedResources = generatedResources;
+        this.compressedMediaTypes = options.getCompressMediaTypes();
+        this.currentClassLoader = Thread.currentThread().getContextClassLoader();
+        this.enableCompression = options.isEnableCompression();
+        this.indexPage = options.getIndexPage();
+        this.enableRangeSupport = options.isEnableRangeSupport();
+    }
+
+    @Override
+    public void handle(RoutingContext context) {
+        if (!ALLOWED_HTTP_METHODS.contains(context.request().method())) {
+            LOG.debugf("HTTP method '%s' not allowed, this static handler accepts only GET, OPTIONS and HEAD",
+                    context.request().method().name());
+            beforeNextHandler(this.currentClassLoader, context);
+        } else {
+            doHandle(context);
+        }
+
+    }
+
+    private String removeStartSlashes(String value) {
+        return value.replaceAll("^/+", "");
+    }
+
+    private void doHandle(RoutingContext context) {
+
+        if (context.request().method().equals(HttpMethod.HEAD)) {
+            context.response().end();
```

Comment:
I think the HEAD method should also return the content size, all headers but not the actual content.